transforms/transforms.go

```diff
@@ -215,3 +215,28 @@ func Unique(hash RowIdentifier) optimus.TransformFunc {
 		return false, nil
 	})
 }
+
+// GroupBy returns a TransformFunc that returns Rows of Rows grouped by id (see tests).
+func GroupBy(identifier RowIdentifier) optimus.TransformFunc {
+	return func(in <-chan optimus.Row, out chan<- optimus.Row) error {
+		groups := map[string][]optimus.Row{}
+		for row := range in {
+			val, err := identifier(row)
+			if err != nil {
+				return err
+			}
+			str, ok := val.(string)
+			if !ok {
+				return fmt.Errorf("groupBy identifier must be a string. was %#v", val)
```

Comment:
I thought it was also because groups was a `map[string][]optimus.Row{}`.  Can we have maps with generic keys?